src/lookup.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _LOOKUP_H
+#define _LOOKUP_H
+
+#include <assert.h>
+#include <jvmti.h>
+#include "arguments.h"
+#include "dictionary.h"
+#include "index.h"
+#include "vmEntry.h"
+
+class MethodInfo {
+  public:
+    bool _mark = false;
+    u32 _key = 0;
+    u32 _class = 0;
+    u32 _name = 0;
+    u32 _sig = 0;
+    jint _modifiers = 0;
+    jint _line_number_table_size = 0;
+    jvmtiLineNumberEntry* _line_number_table = nullptr;
+    FrameTypeId _type;
+
+    jint getLineNumber(jint bci);
+};
+
+class MethodMap : public std::map<jmethodID, MethodInfo> {
```

Comment:
Maybe unordered_map?

Reply:
Doesn't seem like it's used as an ordered map indeed: c7abb85eb9292dd93c07f03a0f26bceec6157103